WebView/app/src/main/java/com/android/samples/webviewdemo/MainActivity.kt

```diff
@@ -54,16 +50,32 @@ class MainActivity : AppCompatActivity() {
         }
     }
 
+    /**
+     * This method checks to see if WebMessageListener is supported. This method is prefered
+     * for security reasons. If this feature is supported, then it is used to create a
+     * Javascript object to be injected into every iframe. If it is not supported then
+     * then the app will fall back to JavascriptInterface.Here, the custom object that is inserted
+     * using JavascriptInterface is setup to mimic the object that would be created by WebMessageListener
+     * if it were available. This is so that the that Javascript will not have to behave
+     * any differently whether it is passed an object by WebMessageListener or JavascriptInterface.
+     * @param webview the component that is
+     * @param mContext the amount of incoming damage
+     * @param jsObjName the name that will be given to the Javascript objects created by either
+     *        WebMessageListener or JavascriptInterface
+     * @param allowedOriginRules a set of rules, a frame must match an Origin in this set to have
```

Comment:
Please mention this is only used on a best-effort basis if the WebMessageListener feature is available